prow/slack/client.go

```
@@ -0,0 +1,156 @@
+	IsGeneral      bool     `json:"is_general"`
+	NameNormalized string   `json:"name_normalized"`
+	IsShared       bool     `json:"is_shared"`
+	IsOrgShared    bool     `json:"is_org_shared"`
+	IsMember       bool     `json:"is_member"`
+	Members        []string `json:"members"`
+	Topic          struct {
+		Value   string `json:"value"`
+		Creator string `json:"creator"`
+		LastSet int    `json:"last_set"`
+	} `json:"topic"`
+	Purpose struct {
+		Value   string `json:"value"`
+		Creator string `json:"creator"`
+		LastSet int    `json:"last_set"`
+	} `json:"purpose"`
+	PreviousNames []interface{} `json:"previous_names"`
+	NumMembers    int           `json:"num_members"`
+}
+
+type ChannelList struct {
+	Ok       bool      `json:"ok"`
+	Channels []Channel `json:"channels"`
+}
+
+// Create a slack client with an API token.
+func NewClient(token string) *Client {
+	return &Client{
+		token: token,
+	}
+}
+
+func (sl *Client) VerifyAPI() bool {
+	t, e := sl.PostMessage(apiTest, sl.urlValues())
+	if e != nil {
+		return false
+	}
+
+	var apiResponse ApiResponse
+	e = json.Unmarshal(t, &apiResponse)
+	return apiResponse.Ok
+}
+
+func (sl *Client) VerifyAuth() bool {
+	t, e := sl.PostMessage(authTest, sl.urlValues())
+	if e != nil {
+		return false
+	}
+
+	var authResponse AuthResponse
+	e = json.Unmarshal(t, &authResponse)
+	return authResponse.Ok
+}
+
+func (sl *Client) urlValues() *url.Values {
+	uv := url.Values{}
+	uv.Add("token", sl.token)
+	return &uv
+}
+
+func (sl *Client) PostMessage(url string, uv *url.Values) ([]byte, error) {
+	resp, err := http.PostForm(url, *uv)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		t, _ := ioutil.ReadAll(resp.Body)
+		return nil, errors.New(string(t))
+	}
+	t, _ := ioutil.ReadAll(resp.Body)
+	return t, nil
+}
+
+func (sl *Client) GetChannels() []Channel {
+	var uv *url.Values = sl.urlValues()
+	t, _ := sl.PostMessage(channelsList, uv)
```

ignored err? and seems `GetChannels()` is not used anywhere?